### PR TITLE
FINERACT-1065: Fixing integration test issues

### PR DESCRIPTION
**Pull request by ptuomola**
## Description
Describe the changes made and why they were made. Ignore if these details are present on the associated Jira ticket

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [x] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [x] API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm has been updated with details of any API changes.

- [x] Integration tests have been created/updated for verifying the changes made.

- [x] All Integrations tests are passing with the new commits.

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

### CODE REVIEW

**Review by awasum — APPROVED** [review]

**Comment by awasum** [comment]
@vorburger , does this look good to you? Merge?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountTransactionRepository.java [hunk 1]
```diff
@@ -21,11 +21,13 @@
...
 import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface SavingsAccountTransactionRepository
         extends JpaRepository<SavingsAccountTransaction, Long>, JpaSpecificationExecutor<SavingsAccountTransaction> {
 
     @Query("select sat from SavingsAccountTransaction sat where sat.id = :transactionId and sat.savingsAccount.id = :savingsId")
-    SavingsAccountTransaction findOneByIdAndSavingsAccountId(Long transactionId, Long savingsId);
+    SavingsAccountTransaction findOneByIdAndSavingsAccountId(@Param("transactionId") Long transactionId,
+            @Param("savingsId") Long savingsId);
```
Comment (awasum) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountTransactionRepository.java:31]: Is there a way to make things like this to fail the build? like a tool that will throw an error saying "you have not passed the parameter a and b". Does this mean, this query was not working before now?

Comment (vorburger) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountTransactionRepository.java:31]: @awasum @ptuomola I think it should work without this - but there is a subtlety: Spring can only find out the name of parameters if code is compiled with `-parameters`; NB e.g. _To store formal parameter names in a particular .class file, and thus enable the Reflection API to retrieve formal parameter names, compile the source file with the -parameters option to the javac compiler._ on https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html. To be able to run in Eclipse, you need to enable https://stackoverflow.com/a/37391164/421602. In an ideal world, we could figure out how to add this to Gradle to have Buildship do that.. or we could keep a `.settings/` in Git.

**Review by vorburger — APPROVED** [review]

**Pull request merged**